Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm

```diff
@@ -327,4 +327,28 @@
     [m_webView didChangeValueForKey:@"_hasActiveNowPlayingSession"];
 }
 
+void PageClientImplCocoa::videoControlsManagerDidChange()
+{
+    [m_webView willChangeValueForKey:@"_canToggleFullscreen"];
+    [m_webView didChangeValueForKey:@"_canToggleFullscreen"];
+}
+
+#if ENABLE(VIDEO_PRESENTATION_MODE)
+void PageClientImplCocoa::didEnterFullscreen()
+{
+    isFullScreenChanged();
+}
+
+void PageClientImplCocoa::didExitFullscreen()
+{
+    isFullScreenChanged();
+}
+
+void PageClientImplCocoa::isFullScreenChanged()
+{
+    [m_webView willChangeValueForKey:@"_isInFullscreen"];
+    [m_webView didChangeValueForKey:@"_isInFullscreen"];
+}
```

Comment:
Should clients be listening to `_isInFullscreen`, or should we have them use this API instead?:

```
@property (nonatomic, readonly) WKFullscreenState fullscreenState WK_API_AVAILABLE(macos(13.0), ios(16.0));
```

Reply:
I didn't know this existed! This is better than using `_isInFullscreen` (and it's already key-value observable).